pkg/portal/graphql/resource.go

```diff
@@ -66,5 +69,31 @@ var appResource = graphql.NewObject(graphql.ObjectConfig{
 				return base64.StdEncoding.EncodeToString(result.([]byte)), nil
 			},
 		},
+		"checksum": &graphql.Field{
+			Type: graphql.String,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				ctx := GQLContext(p.Context)
+				r := p.Source.(*model.AppResource)
+				resMgr := ctx.AppResMgrFactory.NewManagerWithAppContext(r.Context)
+				result, err := resMgr.ReadAppFile(r.DescriptedPath.Descriptor,
+					&resource.AppFile{
+						Path: r.DescriptedPath.Path,
+					})
+				if errors.Is(err, resource.ErrResourceNotFound) {
+					return nil, nil
+				} else if err != nil {
+					return nil, err
+				}
+				// Calculate the checksum with crc32 IEEE
+				checksum := crc32.ChecksumIEEE(result.([]byte))
```

Comment:
Add `pkg/util/checksum/checksum.go` and add `func CRC32IEEEInHex(b []byte) string`. Add some test for it.